### PR TITLE
[image_picker] Fix ios scaling in portrait mode

### PR DESCRIPTION
**Pull request by Xavjer**
## Description

Fixes the issue, that within image_picker when taking a picture on iOS in Portrait mode, the scaling did not work as intended (AspectRatio was flipped)

## Related Issues

https://github.com/flutter/flutter/issues/40334 => Will be fixed with this PR

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
=> No new unit tests added, as the existing should already cover this, but the image did not have the values that were returned from the channel.
- [X] All existing and new tests are passing.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by cyanglaz** [comment]
@xavjer Can we just flip the width and height for portrait pictures instead of using a completely different API? This change may be a bit risky until we set up a CI and write tests for this.

**Comment by cyanglaz** [comment]
I created this PR for an alternative fix. Do you want to take a look? @Xavjer

**Comment by Xavjer** [comment]
@cyanglaz I was thinking about that too, but from my understanding this is the better way to create a thumbnail, as the resulting image has a better resolution (and memory/performance improvement, especially during the case that we have to rotate the image) and it get's rid of the "issue" to rotate the image first.

This is apple's recommended way, see: https://medium.com/@prafullkumar77/image-usage-memory-comparison-and-best-practices-in-ios-wwdc2018-4a8919019ae9
Or the WWDC2018 video referenced there https://developer.apple.com/videos/play/wwdc2018/219/

Your fix should work too, but I was trying to improve the code itself not only fix the issue.

File: packages/image_picker/ios/Classes/FLTImagePickerImageUtil.m [hunk 1]
```diff
@@ -31,57 +31,18 @@ @implementation FLTImagePickerImageUtil : NSObject
...
-
-  UIImage *scaledImage = UIGraphicsGetImageFromCurrentImageContext();
-  UIGraphicsEndImageContext();
-  return scaledImage;
+  NSData *imageData = UIImagePNGRepresentation(image);
+  CGImageSourceRef src = CGImageSourceCreateWithData((__bridge CFDataRef)imageData, NULL);
+  CFDictionaryRef options = (__bridge CFDictionaryRef) @{
+    (id)kCGImageSourceCreateThumbnailWithTransform : @YES,
+    (id)kCGImageSourceCreateThumbnailFromImageAlways : @YES,
+    (id)kCGImageSourceThumbnailMaxPixelSize : MAX(maxWidth, maxHeight)
```
Comment (cyanglaz) [inline packages/image_picker/ios/Classes/FLTImagePickerImageUtil.m]: It seems that the behavior would be different here.
Imagine we have an image that we want to specify maxWidth to be 100 and height to be 400. The original image is very big say 2000 by 4000.
In our old implementation, the final image would be 100 * 200.
In this implementation, the final image would be 400 * 800.

Comment (Xavjer) [reply packages/image_picker/ios/Classes/FLTImagePickerImageUtil.m]: You are right, that would happen in this solution. There is only a maxDimension which can be set with this, therefore your recommended way will probably be the correct solution. For my usecase, I want to reduce the imagesize for the larger side to be 1024, therefore that worked for my case.

**Comment by cyanglaz** [comment]
Closing this PR and we will solve the issue with https://github.com/flutter/plugins/pull/2070
@Xavjer If think your solution is more for https://github.com/flutter/flutter/issues/26191. If you are interested, you could create a PR to add a new method which generates a thumbnail for the image

**Pull request closed without merging**